client/rpmtrans.c

```diff
@@ -211,13 +211,104 @@ TDNFPopulateTransaction(
     goto cleanup;
 }
 
+uint32_t
+TDNFDetectPreTransFailure(
+    rpmts pTS,
+    char *pszError
+    )
+{
+    uint32_t dwError = 0;
+    rpmtsi pi = NULL;
+    rpmte pte = NULL;
+    char *pszToken;
+    int i = 0;
+    char *pszPkgName = NULL;
+    char *pszSymbol = NULL;
+    char *pszVersion = NULL;
+    char *pszEVR = NULL;
+    char *pszCachePkgName = NULL;
+    char *pszCachePkgEVR = NULL;
+
+    // Error Str has the format: <Pkg-Name> <Symbol> <version-release>
+    pszToken = strtok(pszError, " ");
+    while (pszToken != NULL)
+    {
+        switch(i)
+        {
+            case 0:
+                pszPkgName = pszToken;
+                break;
+            case 1:
+                pszSymbol = pszToken;
+                break;
+            case 2:
+                pszVersion = pszToken;
+                break;
+        }
+        i++;
+        pszToken = strtok(NULL, " ");
+    }
+
+    if (!pszPkgName || !pszSymbol || !pszVersion)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pi = rpmtsiInit(pTS);
```

Comment:
we haven't checked if pTS is valid as well? Also, once check if we should validate pi before passing it in rpmtsiNext.

Reply:
Done. Thanks.